### PR TITLE
Nigel/feature/customer

### PR DESCRIPTION
**Pull request by Pocketzs**
This branch has customer and customer repo completed with TDD and spec harness passing all customer tests except the expected updated/created tests
image

image (1)

### CODE REVIEW

File: lib/customer.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    @last_name = attributes[:last_name]
+    @created_at = time_converter(attributes[:created_at])
+    @updated_at = time_converter(attributes[:updated_at])
+  end
+
+  def time_converter(attributes)
+    return unless attributes
+    Time.parse(attributes)
+  end
+end
```
[the remark this answers is not in the document]
Comment (Pocketzs) [reply lib/customer.rb:21]: fixed

File: lib/customer.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+require 'time'
+class Customer
+  attr_accessor :id,
+              :first_name,
+              :last_name,
+              :created_at,
+              :updated_at
+
+  def initialize(attributes, repo = nil)
```
[the remark this answers is not in the document]
Comment (Pocketzs) [reply lib/customer.rb:9]: Will ignore

File: lib/customer.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+require 'time'
+class Customer
+  attr_accessor :id,
+              :first_name,
+              :last_name,
+              :created_at,
```
Comment (Pocketzs) [inline lib/customer.rb]: Fixed

File: spec/customer_repository_spec.rb [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+      fragment = 'on'
+
+      expected = cr.find_all_by_last_name(fragment)
+
+      expect(expected.length).to eq(2)
+      expect(expected.first.class).to eq(Customer)
+    end
+  end
+end
```
Comment (Pocketzs) [inline spec/customer_repository_spec.rb]: fixed

File: spec/customer_repository_spec.rb [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+      expected = cr.find_all_by_first_name(fragment)
+
+      expect(expected.length).to eq(2)
+      expect(expected.first.class).to eq(Customer)
+    end
+  end
+
+  describe '#find_all_by_last_name' do
+    it 'returns an empty array or one or more customers matching the substring fragment last name' do
```
Comment (Pocketzs) [inline spec/customer_repository_spec.rb]: Fixed

File: spec/customer_repository_spec.rb [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+      expect(cr.all[0].id).to eq(1)
+      expect(cr.all[0].first_name).to eq('John')
+      expect(cr.all[0].last_name).to eq('Smith')
+      expect(cr.all[0].updated_at).to be > Time.parse('2012-03-27 14:54:09 UTC')
+    end
+  end
+
+  describe '#find_all_by_first_name' do
+    it 'returns an empty array or one or more customers matching the substring fragment first name' do
```
Comment (Pocketzs) [inline spec/customer_repository_spec.rb]: Fixed

**Review by Kledin85 — APPROVED** [review]
nice

**Review by beddings81 — COMMENTED** [review]
Thanks for fixing the hound errors!

**Review by KaraJoHo — APPROVED** [review]

**Pull request merged**